external/external_custom/dockerfile/test.sh

```diff
@@ -0,0 +1,53 @@
+ #/bin/bash
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+source $(dirname "$0")/test_base_functions.sh
+
+# Set up Java to be used by the external_custom test
+
+if [ -d /java/jre/bin ];then
+	echo "Using mounted Java8"
+	export JAVA_BIN=/java/jre/bin
+	export JAVA_HOME=/java
+	export PATH=$JAVA_BIN:$PATH
+elif [ -d /java/bin ]; then
+	echo "Using mounted Java"
+	export JAVA_BIN=/java/bin
+	export JAVA_HOME=/java
+	export PATH=$JAVA_BIN:$PATH
+else
+	echo "Using docker image default Java"
+	java_path=$(type -p java)
+	suffix="/java"
+	java_root=${java_path%$suffix}
+	export JAVA_BIN="$java_root"
+	$JAVA_BIN/java -version
+	export JAVA_HOME="${java_root%/bin}"
+fi
+
+echo_setup
+
+# export MAVEN_OPTS="-Xmx1g"
+
+cd /flink
+pwd
+echo "Compile and run external_custom tests"
+mvn clean install
```

Comment:
This is the EXTERNAL_TEST_CMD variable which should be passed in for cases where its not mvn clean install (which we can use as the default command, as its used most often by applications testing).

Reply:
For this  we have to add the properties in the test.properties of the individual tests and get the values from the test.properties of the test being run and use it in the common_functions.sh, according to my understanding. Is this correct?

Reply:
yes, that is what I was thinking we do, for tests in category 1 (unsuitable for external_custom), you do not define test_command in test.properties (because there are many commands), for those, we need to keep their Dockerfile/xx-test.sh file which contains all the special set up, for tests in category 2 (using mvn clean install) or category 3 (some other simple one line command) those can be added to test_command in test.properties file

for the contents of external_custom/test.properties, then test_command="${EXT_TEST_CMD}" if EXT_TEST_CMD is defined, if not, we use the default value 'mvn clean install'